src/directory_listing.rs

```diff
@@ -435,30 +434,34 @@ fn html_auto_index<'a>(
     );
 
     // Prepare table row template
-    let mut table_row = String::new();
-    if base_path != "/" {
-        table_row = String::from(r#"<tr><td colspan="3"><a href="../">../</a></td></tr>"#);
-    }
+    let table_rows = if base_path == "/" {
+        String::new()
+    } else {
+        String::from(r#"<tr><td colspan="3"><a href="../">../</a></td></tr>"#)
+    };
 
-    for entry in entries {
-        let file_name = &entry.name_encoded;
+    let table_rows = entries.iter().fold(table_rows, |mut output, entry| {
```

Comment:
The code looks much more readable than the previous one. However, did you know if the iterator version here performs better in general than a simple for loop? Maybe is something that we have to double-check (benchmark/profile).

Reply:
Iterators generally have exactly the same performance as for loops in Rust, the optimizer takes care of it. The difference here is not the loop but how the resulting string is being constructed.